addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionManager.java

```diff
@@ -500,91 +500,108 @@ private PathsPromoteResult runPathPromotions( final PathsPromoteRequest request,
-            {
-                logger.info( "Running promotions from: {} (key: {})\n  to: {} (key: {})", sourceStore, request.getSource(),
-                             targetStore, request.getTarget() );
+                if ( errors.isEmpty() )
+                {
+                    logger.info( "Running promotions from: {} (key: {})\n  to: {} (key: {})", sourceStore,
+                                 request.getSource(), targetStore, request.getTarget() );
 
-                final boolean purgeSource = request.isPurgeSource();
-                contents.forEach( (transfer)->{
-                    final String path = transfer.getPath();
+                    final boolean purgeSource = request.isPurgeSource();
+                    contents.forEach( ( transfer ) -> {
+                        final String path = transfer.getPath();
 
-                    if ( !transfer.exists() )
-                    {
-                        pending.remove( path );
-                        skipped.add( path );
-                    }
-                    else
-                    {
-                        try
+                        if ( !transfer.exists() )
                         {
-                            Transfer target = contentManager.getTransfer( targetStore, path, TransferOperation.UPLOAD );
-                            //                        synchronized ( target )
-                            //                        {
-                            // TODO: Should the request object have an overwrite attribute? Is that something the user is qualified to decide?
-                            if ( target != null && target.exists() )
-                            {
-                                logger.warn( "NOT promoting: {} from: {} to: {}. Target file already exists.", path,
-                                             request.getSource(), request.getTarget() );
-
-                                // TODO: There's no guarantee that the pre-existing content is the same!
-                                pending.remove( path );
-                                skipped.add( path );
-                            }
-                            else
+                            pending.remove( path );
+                            skipped.add( path );
+                        }
+                        else
+                        {
+                            try
                             {
-                                try (InputStream stream = transfer.openInputStream( true ))
+                                Transfer target =
+                                        contentManager.getTransfer( targetStore, path, TransferOperation.UPLOAD );
+                                //                        synchronized ( target )
+                                //                        {
+                                // TODO: Should the request object have an overwrite attribute? Is that something the user is qualified to decide?
```

Comment:
I think overwrite should not be allowed. The user could always delete the pre-existing target and resume the promotion, if they are authorized.